File: QuickFind and QuickUnion/quickfind.py
class QuickFind:

    id = []

    def __init__(self, N):
        for x in range(0, N):
            self.id.append(x)  # initialize array

    def union(self, p, q):
        pval = self.id[p]  # value of p
        qval = self.id[q]  # value of q

        for i in range(0, len(self.id)-1):  # loops through id[]
            if self.id[i] == pval:  # if the value at that index is equal to the value of p
                self.id[i] = qval  # change the value at that index to q

    def connected(self, p, q):
        return self.id[p] == self.id[q]

if __name__ == '__main__':
    qf = QuickFind(10)
    qf.union(1, 3)
    qf.union(3, 4)
    qf.union(4, 9)
    print(qf.connected(1, 9))

File: QuickFind and QuickUnion/QuickFind.java
public class QuickFind{

    public static void main(String[] args){
        QF qf = new QF(10);
        qf.union(3,4);
        qf.union(6,9);
        qf.union(2,1);
        System.out.println("6 and 9 connected? " + qf.connected(6,9));
    }

}

class QF {

    private int[] id;

    public QF(int N){
        id = new int[N];
        for(int i = 0; i < N; i++){
            id[i] = i;
        }
    }

    public void union(int p, int q){
        int pval = id[p];
        int qval = id[q];
        for(int i = 0; i < id.length; i++){
            if (id[i] == pval) id[i] = qval;
        }
    }

    public boolean connected(int p, int q){
        return (id[p] == id[q]);
    }

}
